### PR TITLE
increase length of sequence_table column in sales_sequence_meta table

### PR DESCRIPTION
**Pull request by jhruehl**
### Description

Having a too small length for the sequence_table column can lead to integrity constraint errors on setup:upgrade, because longer table names will be cut off (at least the store id suffix of those sales sequence table names).
In our case it was the PostfinanceCW extension, which lead to the tables "sequence_postfinancecw_transaction_0", "sequence_postfinancecw_transaction_1",... and so on (for each store id one table), which are 36 characters long.
So the names for the admin store view (id 0) got cut off in the sales_sequence_meta table and as a consequence lead to an integrity constrained, when the recurring script in the Magento_SalesSequence module tries to add the meta data for the sequence_postfinancecw_transaction_1 table.

So a character length of only 32 is definitely not long enough. I guess 50 character should suffice though. ¯\_(ツ)_/¯

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=17387) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @jhruehl. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

**Review by orlangur — CHANGES_REQUESTED** [review]
Why those table names are not passed through `getTableName` performing a truncation + some MD5 addition?

**Comment by jhruehl** [comment]
@orlangur Just change the field to text or increase the length to 256. The number of rows in this table will probably never come near 200 in whatever reasonable setup. And even with 1000 rows you'll have to severly misconfigure your database for that to have a perfomance impact in any way.

Furthermore hashing (or adding a hash) may seem to be a solution, but it's not good for readability.

But if you still object, feel free to think of a solution, which includes 32 characters, where the important information is still readable and which doesn't lead to problems or has to be overhauled in the long run.

**Comment by orlangur** [comment]
@jhruehl I mean there is already a mechanism to not have too long table names, just wondering why it wasn't used here.

**Comment by sidolov** [comment]
Hi @jhruehl , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Comment by jhruehl** [comment]
@sidolov Actually I've done my part and offered a solution, which doesn't hurt and just increases the length of a column in a table, which will never reach a size where this could have an impact.

And because I guess no one on your side has the time or is willing to do react to orlangur's question, I'm definitely not interested (not my job), I guess it's better to initially accept my pull request than letting the issue be lost in nothingness over time and having people raising questions every now and then.

To summarize my opinion in more clear word: "It's just an increase of a stupid column lenght value, which type should have been text anyway." Hier wird aus einer Mücke ein Elefant gemacht.

**Comment by heldchen** [comment]
ran into this today. sequence name is silently truncated by myslq, thus getting 
```
  SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'postfinancecw_transaction-0' for key 'SALES_SEQUENCE_META_ENTITY_TYPE_STORE_ID', query was: INSERT INTO `sales_sequence_meta` (`entity_type`, `store_id`, `sequen
  ce_table`) VALUES (?, ?, ?)
```
during a `setup:db-schema:upgrade` as sequence table is truncated to 32 chars. this pull request should have been accepted.

**Comment by orlangur** [comment]
@jhruehl it was determined in issue you reported that correct value is 64 characters.

Could you please change this value and amend your commit?

**Comment by jhruehl** [comment]
@orlangur I've changed it to 64, ran into a merge conflict and corrected it by comparing the given version of the db_schema.xml file in the current 2.3.0-develop branch. Hope this suffices.

**Comment by orlangur** [comment]
@jhruehl oh, now there are four commits :) Can you please squash into single one and force push? I can assist with that if needed.

**Comment by orlangur** [comment]
@jhruehl oh, now there are four commits :) Can you please squash into single one and force push? I can assist with that if needed.

**Review by VladimirZaets — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, thank you for the review.
[ENGCOM-3757](https://jira.corp.magento.com/browse/ENGCOM-3757) has been created to process this Pull Request

**Comment by orlangur** [comment]
@jhruehl could you please update branch as requested?

@VladimirZaets it is not a good idea to update a number in few consecutive commits.

**Comment by p-bystritsky** [comment]
@orlangur done.

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @jhruehl. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.1. 
Please check the release notes for final confirmation.
